pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -210,13 +210,65 @@ func (adc *attachDetachController) Run(stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
 	glog.Infof("Starting Attach Detach Controller")
 
+	err := adc.populateActualStateOfWorld()
+	if err != nil {
+		glog.Errorf("error populating the actual state of world: %v", err)
+	}
+	err = adc.populateDesiredStateOfWorld()
+	if err != nil {
+		glog.Errorf("error populating the desired state of world: %v", err)
+	}
 	go adc.reconciler.Run(stopCh)
 	go adc.desiredStateOfWorldPopulator.Run(stopCh)
 
 	<-stopCh
 	glog.Infof("Shutting down Attach Detach Controller")
 }
 
+func (adc *attachDetachController) populateActualStateOfWorld() error {
+	glog.V(5).Infof("Populating ActualStateOfworld")
+	nodes, err := adc.kubeClient.Core().Nodes().List(v1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for i := range nodes.Items {
+		nodeName := types.NodeName(nodes.Items[i].Name)
+		for _, attachedVolume := range nodes.Items[i].Status.VolumesAttached {
+			uniqueName := string(attachedVolume.Name)
+			pluginName, volumeName := volumehelper.SplitUniqueName(uniqueName)
+			plug, err := adc.volumePluginMgr.FindPluginByName(pluginName)
+			if err != nil {
+				glog.Errorf("could not find plugin named %s for volume %s: %v", pluginName, volumeName, err)
+				continue
+			}
+			volumeSpec, err := plug.ConstructVolumeSpecFromName(volumeName)
```

Comment:
ConstructVolumeSpecFromName function fake volumeSource information for almost all of volume plugins. The main issue is that this information will be used by VerifyVolumesAreAttached(). If this information (e.g., volumeID) is wrong, the function will return wrong information related whether volumes are attached or not. 

One way to avoid this is populate desired state first which contains this information, and then find the matching volume to populate this spec information from desired state. But volume spec information might not be recovered if pod is deleted so desired state does not have it either. PR #40239 might help solve this issue, but this PR is revered. Will follow up with the status.